### PR TITLE
enhance(workspace): Create new store interfaces and NoteStore

### PR DESCRIPTION
**Pull request by tma66**
**Background**
This is the first step towards the Dendron store refactor. We are working towards a store layer in charge of writing/retrieving data along with its metadata while moving other business out (and into the engine layer). This involves creating a new metadatastore to interact with metadata and a new filestore to interact with a filesystem.

See https://docs.google.com/document/d/1iDGJG65KZqrBM_Ib_xOsMXTF5FC3e8eoF_lGykZmLhs/edit#heading=h.9datrr3u246k

**Changes**
1. Create interfaces for NoteStore which consists of a filestore and metadatastore. This includes the basic methods that our store will support but is not necessarily the final. We will add more as needed
2. NoteStore with CRUD operations implemented aside from `rename`. This class will be in charge of retrieving and writing to both the filestore and metadatastore.
3. Update serialization of NoteProps to file so that it does not append a "\n" to the body. The `matter` library modifies the body by appending a "\n" but this is inconsistent with the theme that what we write should equal what we retrieve.

**Test**
1. Add new test file to test `NoteStore.ts`

**Up next**
1. Integrate NoteStore with a new engine

### CODE REVIEW

**Review by bpathakota — APPROVED** [review]
commit message looks good!

File: packages/engine-test-utils/src/__tests__/engine-server/store/noteStore.spec.ts [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+  DendronEngineClient,
+  NodeJSFileStore,
+  NoteMetadataStore,
+  NoteStore,
+} from "@dendronhq/engine-server";
+import _ from "lodash";
+import { runEngineTestV5 } from "../../../engine";
+import { ENGINE_HOOKS } from "../../../presets";
+
+describe("NoteStore", () => {
```
Comment (kevinslin) [inline packages/engine-test-utils/src/__tests__/engine-server/store/noteStore.spec.ts:13]: can you write the test in the test style? [[Writing Tests|dendron://dendron.docs/dev.process.qa.test#writing-tests]]

GIVEN "NoteStore", WHEN running note.find, THEN ...

Comment (kevinslin) [reply packages/engine-test-utils/src/__tests__/engine-server/store/noteStore.spec.ts:13]: same comment for other tests

File: packages/engine-server/src/store/NodeJSFileStore.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  RespV3,
+} from "@dendronhq/common-all";
+import _ from "lodash";
+import { getAllFiles } from "@dendronhq/common-server";
+
+export class NodeJSFileStore implements IFileStore {
+  /**
+   * See {@link IFileStore.read}
+   */
+  async read(fpath: string): Promise<RespV3<string>> {
```
[the remark this answers is not in the document]
Comment (tma66) [reply packages/engine-server/src/store/NodeJSFileStore.ts:17]: Sure, I'll change to URI

File: packages/engine-server/src/store/NoteStore.ts [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+  private _fileStore: IFileStore;
+  private _metadataStore: IDataStore<string, NotePropsMeta>;
+  private _wsRoot: string;
+  private _logger: DLogger;
+
+  constructor(
+    fileStore: IFileStore,
+    dataStore: IDataStore<string, NotePropsMeta>,
+    wsRoot: string,
+    logger: DLogger
```
[the remark this answers is not in the document]
Comment (tma66) [reply packages/engine-server/src/store/NoteStore.ts:38]: I'll follow current usages of `createDisposableLogger` and just not pass in a logger, but instead create disposable logger instead.

**Review by kevinslin — APPROVED** [review]

**Pull request merged**